src/stan/services/pathfinder/single.hpp

```diff
@@ -820,7 +820,16 @@ inline auto pathfinder_lbfgs_single(
         logger.info(lbfgs_ss);
         lbfgs_ss.str("");
       }
-      throw e;
+      if (ReturnLpSamples) {
+        // we want to terminate multi-path pathfinder during these unrecoverable
+        // exceptions
+        throw;
+      } else {
+        logger.error(e.what());
```

Comment:
`path_num` should be appended to this message?

Reply:
If `ReturnLPSamples` is false, we're always in single path pathfinder, right? If so, I don't think the path number is important

Reply:
Whoops! Wrong order that is correct